src/main/java/http_server/RequestParser.java

```diff
@@ -0,0 +1,82 @@
+package http_server;
+
+import java.util.HashMap;
+
+public class RequestParser {
+    HashMap<String, String> requestCollection = new HashMap<>();
+    HashMap<String, String> headersCollection = new HashMap<>();
+
+    public RequestParser(String request) {
+        String[] splitRequest = request.trim().split("\r\n");
+
+        responseLineParser(splitRequest);
+        bodyParser(splitRequest);
+        headerParser(splitRequest);
+    }
+
+    public String method() {
```

Comment:
This method and the body and route methods are coupled to the responseLineParser and bodyParser methods. As a result, this implementation is unclear - (`"method"`) appears to come out of nowhere. Can you think of a clearer way to parse requests and get each component of a request? Perhaps you're missing an abstraction...